File: src/py_logic/metrics.py
async def get_metrics(metrics, user_data):
    data = {}

    if not metrics or len(metrics) == 0:
        return data

    await asyncio.gather(*[
        asyncio.create_task(assign_data(metric, data, user_data))
        for metric in metrics
    ])

    return data

async def assign_data(metric, data, user_data):
    data[metric['key']] = user_data[metric['key']]

File: node_logic/src/logic/metrics.js
async function getMetrics(metrics, userData) {
    let data =  {

    };

    if (!metrics || metrics.length == 0) {
        return data;
    }

    await Promise.all(
        metrics.map(async metric => {
            data[metric.key] = userData[metric.key];
        })
    );

    return data;
}

module.exports = { getMetrics }